scripts/build/collect_symbols.sh

```diff
@@ -26,43 +36,69 @@ done
 
 if [[ -z "${PLATFORM}" ]]; then
   echo "MUST SPECIFY PLATFORM"
+  usage
+  exit 1
+fi
+
+if [[ -z "${CRATE_NAME}" ]]; then
+  echo "MUST SPECIFY CRATE_NAME"
+  usage
   exit 1
 fi
 
 REPO_ROOT="$(git rev-parse --show-toplevel)"
-AWS_LC_DIR="${REPO_ROOT}/aws-lc-sys/aws-lc"
-SYMBOLS_FILE="${REPO_ROOT}/aws-lc-sys/symbols/${PLATFORM}.txt"
+AWS_LC_DIR="${REPO_ROOT}/${CRATE_NAME}/aws-lc"
+SYMBOLS_FILE="${REPO_ROOT}/${CRATE_NAME}/symbols/${PLATFORM}.txt"
 
+if [[ ! -d "${AWS_LC_DIR}" ]]; then
+  echo "INVALID DIRECTORY: ${AWS_LC_DIR}"
+  usage
+  exit 1
+fi
 
 function filter_symbols() {
-    grep -v "^_\?bignum_" | grep -v "_\?curve25519_x25519" | grep -v "_\?edwards25519_"
+  grep -v -E "^bignum_" | grep -v "curve25519_x25519" | grep -v "edwards25519_"
+}
+
+function filter_nm_symbols() {
+  grep -v -E '^_Z' | grep -v 'BORINGSSL_bcm_' | grep -v 'BORINGSSL_integrity_test'
+}
+
+function filter_macho_symbols() {
+  grep -E '^_' | sed -e 's/^_\(.*\)/\1/'
 }
 
 function find_libcrypto() {
-  find "${REPO_ROOT}/target" -type f -name "libcrypto.a" | grep "aws-lc-sys"
+  find "${REPO_ROOT}/target" -type f \( -name "lib*crypto.a" -o -name "lib*crypto.so" -o -name "lib*crypto.dylib" \) | grep "${CRATE_NAME}"
```

Comment:
Actually now that I think about it this technically doesn't filter it down to one directory with the grep. As the target directories will have the crate name with a suffix of some sort of hash.

Reply:
Correct.  This script would likely only succeed after a "clean" build was performed.